src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -431,13 +433,32 @@ protected Task ResolveAuthorityAsync()
                 tokenClient.AddHeaderToClient(CcsHeader.Value.Key, CcsHeader.Value.Value);
             }
 
+            InjectPcaSsoPolicyHeader(tokenClient);
+
             return tokenClient.SendTokenRequestAsync(
                 additionalBodyParameters,
                 scopes,
                 tokenEndpoint,
                 cancellationToken);
         }
 
+        private void InjectPcaSsoPolicyHeader(TokenClient tokenClient)
```

Comment:
@ashok672 - I played a bit with the code and I think this is better place for the injection of the header. Let me know what you think. If you disagree, just revert my commit and we can discuss alternatives. 

OAuth2Client is too much in the hot path.

Reply:
I do like this. I previously forgot to check if policy is enabeld in this particular flow and I added that on top of your change.